src/Microsoft.Diagnostics.Monitoring.HostingStartup/ParameterCapturing/ObjectFormatting/Formatters/DebuggerDisplay/ExpressionBinder.cs

```diff
@@ -0,0 +1,234 @@
+        {
+            //
+            // NOTE: Complex expressions are not supported (e.g. "Count + 1").
+            // This includes expressions with additional parenthesis that are otherwise supported, e.g. "(Count)".
+            //
+            // The following expressions are supported:
+            // - A property.
+            // - A field.
+            // - A method without parameters.
+            //
+
+            try
+            {
+                if (expression.EndsWith("()", StringComparison.Ordinal))
+                {
+                    return BindMethod(objType, expression[..^2]);
+                }
+
+                return BindProperty(objType, expression) ?? BindField(objType, expression);
+            }
+            catch
+            {
+            }
+
+            return null;
+        }
+
+        private static ExpressionEvaluator? BindMethod(Type objType, string methodName)
+        {
+            MethodInfo? method = objType.GetMethod(
+                methodName,
+                BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Instance | BindingFlags.Static,
+                Array.Empty<Type>());
+
+            if (method == null)
+            {
+                return null;
+            }
+
+            return new ExpressionEvaluator(
+                method.HasImplicitThis()
+                    ? (obj) => method.Invoke(obj, parameters: null)
+                    : (_) => method.Invoke(obj: null, parameters: null),
+                method.ReturnType);
+        }
+
+        private static ExpressionEvaluator? BindProperty(Type objType, string propertyName)
+        {
+            //
+            // Use the property's Get method instead of directly using ".GetValue(..)"
+            // so that we can identify if it's static or not.
+            //
+            MethodInfo? getterMethod = objType.GetProperty(propertyName,
+                BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Instance | BindingFlags.Static
+                )?.GetGetMethod(nonPublic: true);
+
+            if (getterMethod == null)
+            {
+                return null;
+            }
+
+            return new ExpressionEvaluator(
+                getterMethod.HasImplicitThis()
+                    ? (obj) => getterMethod.Invoke(obj, parameters: null)
+                    : (_) => getterMethod.Invoke(obj: null, parameters: null),
+                getterMethod.ReturnType);
+        }
+
+        private static ExpressionEvaluator? BindField(Type objType, string fieldName)
+        {
+            FieldInfo? field = objType.GetField(fieldName,
+                BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Instance | BindingFlags.Static);
+
+            if (field == null)
+            {
+                return null;
+            }
+
+            return new ExpressionEvaluator(
+                field.IsStatic
+                    ? (_) => field.GetValue(obj: null)
+                    : field.GetValue,
```

Comment:
Doesn't this need the `obj` to be passed in order to get the value of an instance field?